File: 찬미/Greedy/11047.swift
let num = readLine()!.split(separator: " ").map { Int($0)! }
let n = num[0]
var k = num[1]

var coins: [Int] = []
var cnt = 0

for _ in 0..<n {
    coins.append(Int(readLine()!)!)
}

coins = coins.reversed()

for coin in coins {
    if coin <= k {
        cnt += (k / coin)
        k %= coin
    }
}

print(cnt)

File: 한유진/11047.py
n , k = map(int,input().split())
coin = list()
for _ in range(n):
    coin.append(int(input()))
count = 0
i = -1
while k > 0 :
    c =  k // coin[i]
    k = k % coin[i]
    if c > 0:
        count += c
    i -= 1
print(count)